realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -585,6 +622,7 @@ private void emitCopyOrUpdateMethod(JavaWriter writer) throws IOException {
                 .beginControlFlow("if (rowIndex != TableOrView.NO_MATCH)")
                     .emitStatement("realmObject = new %s()", Utils.getProxyClassName(className))
                     .emitStatement("realmObject.realm = realm")
+                    .emitStatement("realmObject.updateColumnInfo(realm.getColumnInfo(%s.class))", className)
```

Comment:
why have a `updateColumnInfo` method on `RealmObject` when it already has a reference to `realm` which has the information?